### DISCUSSION TITLE
Preserving data integrity demo

### DISCUSSION THREAD

**Discussion by sankalpk** [electric-sql/electric#560] [Bug reports]
I was able to create an inconsistent state in the "Preserving data integrity" demo.

![Xnip2023-09-30_12-23-56](https://github.com/electric-sql/electric/assets/1317291/e41be10c-3c8f-4045-b379-199100fc3b60)

Note sure whether it's a bug or not, but "Try it out for yourself. Can you cause an integrity violation?" seems to indicate that the demo might be resilient to these sorts of issues.

Steps to reproduce:
1. Turn off connectivity on the left panel.
2. On the right panel, add 1 color to tournament 1
3. On the left panel, add 1 color to tournament 2
4. Turn on connectivity on the left panel.
5. Note that they don't sync to the same state.

**Comment by gorbak25** [comment]
@sankalpk I was unable to replicate the problem using the steps u described. On the other hand after fiddling a lot I managed to also break it:
![image](https://github.com/electric-sql/electric/assets/1352848/66980906-67c1-4b47-a49b-c7baca4cb603)
but it looks like a bug in the js client not the protocol itself
![image](https://github.com/electric-sql/electric/assets/1352848/282241b9-b953-4b6f-b13a-f27ecc86c5f1)

**Comment by balegas** [comment]
Thanks for reporting this. We'll be looking into it shortly.

**Comment by balegas** [comment]
Moving this into a [bug report](https://github.com/electric-sql/electric/discussions/categories/bug-reports) discussion. We will open an issue once we have a way of reproducing it.

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
